pkg/binding/definition.go

```diff
@@ -83,26 +84,26 @@ type stringFromDataFieldDefinition struct {
 	secretConfigMapReader *secretConfigMapReader
 	objectType            objectType
 	outputName            string
-	path                  []string
-	sourceKey             string
+	definition
+	sourceKey string
 }
 
 var _ Definition = (*stringFromDataFieldDefinition)(nil)
 
-func (d *stringFromDataFieldDefinition) GetPath() []string { return d.path }
-
 func (d *stringFromDataFieldDefinition) Apply(u *unstructured.Unstructured) (Value, error) {
 	if d.secretConfigMapReader == nil {
 		return nil, errors.New("kubeClient required for this functionality")
 	}
 
-	resourceName, ok, err := unstructured.NestedString(u.Object, d.path...)
+	res, err := getValuesByJSONPath(u.Object, d.path)
 	if err != nil {
 		return nil, err
 	}
-	if !ok {
-		return nil, errors.New("not found")
+
+	if len(res) != 1 {
+		return nil, fmt.Errorf("Only one value should be returned for %v but we got %v", d.path, res)
```

Comment:
Error strings should not be capitalized.
Ref. https://github.com/golang/go/wiki/CodeReviewComments#error-strings